rest/changestest/changes_collection_test.go

```diff
@@ -191,6 +194,130 @@ func TestMultiCollectionChangesUser(t *testing.T) {
+	rest.RequireStatus(t, response, 201)
+	_ = rt.WaitForPendingChanges()
+
+	var changes struct {
+		Results  []db.ChangeEntry
+		Last_Seq db.SequenceID
+	}
+
+	// Issue changes request.  Will initialize cache for channels, and return docs via queries
+	changesResponse := rt.SendUserRequest("GET", "/{{.keyspace1}}/_changes?since=0", "", "bernard")
+	err = base.JSONUnmarshal(changesResponse.Body.Bytes(), &changes)
+	assert.NoError(t, err, "Error unmarshalling changes response")
+	require.Len(t, changes.Results, 1)
+	logChangesResponse(t, changesResponse.Body.Bytes())
+
+	changesResponse = rt.SendUserRequest("GET", "/{{.keyspace2}}/_changes?since=0", "", "bernard")
+	err = base.JSONUnmarshal(changesResponse.Body.Bytes(), &changes)
+	assert.NoError(t, err, "Error unmarshalling changes response")
+	require.Len(t, changes.Results, 0)
+	logChangesResponse(t, changesResponse.Body.Bytes())
+
+	response = rt.SendAdminRequest("PUT", "/{{.keyspace1}}/brn2_c1", `{"value":1, "channels":["BRN"]}`)
+	rest.RequireStatus(t, response, 201)
+	response = rt.SendAdminRequest("PUT", "/{{.keyspace1}}/chr2_c1", `{"value":1, "channels":["CHR"]}`)
+	rest.RequireStatus(t, response, 201)
+	response = rt.SendAdminRequest("PUT", "/{{.keyspace2}}/brn2_c2", `{"value":1, "channels":["BRN"]}`)
+	rest.RequireStatus(t, response, 201)
+	response = rt.SendAdminRequest("PUT", "/{{.keyspace2}}/chr2_c2", `{"value":1, "channels":["CHR"]}`)
+	rest.RequireStatus(t, response, 201)
+	_ = rt.WaitForPendingChanges()
+
+	changesResponse = rt.SendUserRequest("GET", "/{{.keyspace1}}/_changes?since=0", "", "bernard")
+	err = base.JSONUnmarshal(changesResponse.Body.Bytes(), &changes)
+	assert.NoError(t, err, "Error unmarshalling changes response")
+	require.Len(t, changes.Results, 2)
+	logChangesResponse(t, changesResponse.Body.Bytes())
+	lastSeqBRN := changes.Last_Seq
+
+	response = rt.SendAdminRequest("PUT", "/{{.keyspace1}}/brn3_c1", `{"value":1, "channels":["BRN"]}`)
+	rest.RequireStatus(t, response, 201)
+	_ = rt.WaitForPendingChanges()
+
+	changesResponse = rt.SendUserRequest("GET", fmt.Sprintf("/{{.keyspace1}}/_changes?since=%s", lastSeqBRN), "", "bernard")
+	err = base.JSONUnmarshal(changesResponse.Body.Bytes(), &changes)
+	assert.NoError(t, err, "Error unmarshalling changes response")
+	require.Len(t, changes.Results, 1)
+	logChangesResponse(t, changesResponse.Body.Bytes())
+
+	changesResponse = rt.SendUserRequest("GET", "/{{.keyspace1}}/_changes?since=0", "", "charlie")
+	err = base.JSONUnmarshal(changesResponse.Body.Bytes(), &changes)
+	assert.NoError(t, err, "Error unmarshalling changes response")
+	require.Len(t, changes.Results, 2)
+	logChangesResponse(t, changesResponse.Body.Bytes())
+
+	// Start a continuous changes feed in its own goroutine
+	continuousResponse := rt.SendUserRequest("GET", "/{{.keyspace1}}/_changes?feed=continuous&timeout=5000", "", "bernard")
+	reader := bufio.NewReader(continuousResponse.Body)
```

Comment:
I'd be inclined to shorten the timeout here and move this code to the end of the test, where we run a matrix of tests

| user        | keyspace1 | keyspace2 |
| --------  | ---------- | ----------- | 
| bernand |                    |                    |
| charlie    |                    |                    |
| BRNchr  |                    |                    | 

You should be able to run a function like 

```
response := rt.SendUserRequest(http.MethodGet, fmt.Sprintf("/%s/_changes?feed=continuous&since=%s&timeout=2000", keyspace since), "", "bernard")
changes, err := readContinuousChanges(changesResponse)
// Assert number of changes
```

I'd see if we can shorten the 2000 timeout, but we might have to do something fancier if this fails under CBS. The goal is to make the changes feed run as fast as possible without actually failing the test. The changes will be in memory since we have just run the changes feed with one shot, so I think we can drop to 500/1000. In the event that this doesn't work, we'd write an assertion where we re-run the request with increasingly long timeouts until it passes (or fails). But I'm not sure we have to go there yet.

Reply:
https://github.com/couchbase/sync_gateway/pull/6073/commits/2d3706629b8aa3313fac3191cdacb93da1dc08c6
Im not sure if this makes the continuous changes checks a t l.260 and after redundant, but let me know if i should remove anything

Reply:
I think this will make the continuous changes checks redundant and we can remove a bunch of code.

Reply:
Let's remove all this code with `GetNextContinuousChange` and the callers and focus on the `RequireContinuousFeedChangesCount` and that's it for this review.